ods_ci/tests/Resources/Files/pipeline-samples/v2/pip_index_url/flip_coin_pip_index_url_compiled.yaml

```diff
@@ -277,7 +277,7 @@ deploymentSpec:
         - "\nif ! [ -x \"$(command -v pip)\" ]; then\n    python3 -m ensurepip ||\
           \ python3 -m ensurepip --user || apt-get install python3-pip\nfi\n\nPIP_DISABLE_PIP_VERSION_CHECK=1\
           \ python3 -m pip install --quiet --no-warn-script-location --index-url $PIP_INDEX_URL\
-          \ --trusted-host $PIP_TRUSTED_HOST 'kfp==2.7.0' '--no-deps' 'typing-extensions>=3.7.4,<5;\
+          \ --trusted-host $PIP_INDEX_URL 'kfp==2.9.0' '--no-deps' 'typing-extensions>=3.7.4,<5;\
```

Comment:
This is wrong. In flip_coin_pip_index_url.py you need to declare pip_trusted host for each component, so when compiling the pipeline the resulting yaml file will have `--trusted-host $PIP_TRUSTED_HOST `

Something like this (for each component):
```
@dsl.component(base_image=common_base_image,
                            pip_index_urls=['$PIP_INDEX_URL'],
                            pip_trusted_host=['$PIP_TRUSTED_HOST'])
def random_num(low: int, high: int) -> int:
    """Generate a random number between low and high."""
    import random
    result = random.randint(low, high)
    print(result)
    return result
```

This change should be done for all pipelines setting pip_index_url

Reply:
Done